torchdynamo/_eval_frame.c

```diff
@@ -390,52 +388,55 @@ static PyObject *_custom_eval_frame_run_only(PyThreadState *tstate,
   }
 }
 
+static int active_dynamo_threads = 0;
+
+static pthread_mutex_t dynamo_frame_swap_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Will the python GIL make it so we don't need this?  (At least for currently released versions of Python.)